### PR TITLE
Typescript info

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: server/chat-commands/room-settings.ts [hunk 1]
```diff
@@ -1299,6 +1299,52 @@ export const commands: ChatCommands = {
 	removeroomaliashelp: [
 		`/removeroomalias [alias] - removes the given room alias of the room the command was entered in. Requires: & ~`,
 	],
+
+	resettierdisplay: 'roomtierdisplay',
+	roomtierdisplay(target, room, user, connection, cmd) {
```
Comment (Zarel) [inline server/chat-commands/room-settings.ts:1304]: Was this a mistake? It should probably go in a different PR than a TS refactor.

Comment (KrisXV) [reply server/chat-commands/room-settings.ts:1304]: I asked you about this in #sim-dev and you said it was fine, unless I misinterpreted something.

File: server/chat-commands/info.ts [hunk 2]
```diff
@@ -190,21 +190,21 @@ const commands = {
 		}
 
 		const gameRooms = [];
-		for (const room of Rooms.rooms.values()) {
-			if (!room.game) continue;
-			if ((targetUser.id in room.game.playerTable && !targetUser.inRooms.has(room.roomid)) ||
-				(room.auth && room.auth[targetUser.id] === Users.PLAYER_SYMBOL)) {
-				if (room.isPrivate && !canViewAlts) {
+		for (const r of Rooms.rooms.values()) {
```
Comment (Zarel) [inline server/chat-commands/info.ts:193]: ?

Why was this renamed?

Comment (Zarel) [reply server/chat-commands/info.ts:193]: Oh, variable shadowing.

How about `curRoom`?

File: server/chat-commands/info.ts [hunk 3]
```diff
@@ -213,8 +213,8 @@ const commands = {
 			if (punishments.length) {
 				buf += `<br />Room punishments: `;
 
-				buf += punishments.map(([room, punishment]) => {
-					const [punishType, punishUserid, expireTime, reason] = punishment;
+				buf += punishments.map(([r, p]) => {
```
Comment (Zarel) [inline server/chat-commands/info.ts:216]: ditto here (`curRoom`, `punishment`)

Comment (KrisXV) [reply server/chat-commands/info.ts:216]: punishment is a shadowed variable too, any ideas for that

File: server/chat-commands/info.ts [hunk 4]
```diff
@@ -483,7 +489,7 @@ const commands = {
 		const targetId = toID(target);
 		if (!targetId) return this.parse('/help data');
 		const targetNum = parseInt(target);
-		if (!isNaN(targetNum) && '' + targetNum === target) {
+		if (!isNaN(targetNum) && String(targetNum) === target) {
```
Comment (Zarel) [inline server/chat-commands/info.ts:492]:
```suggestion
		if (!isNaN(targetNum) &&`${targetNum}` === target) {
```

File: server/chat-commands/info.ts [hunk 5]
```diff
@@ -517,30 +522,36 @@ const commands = {
...
+			let details: {[k: string]: string} = {};
 			switch (newTarget.searchType) {
 			case 'nature':
 				const nature = Dex.getNature(newTarget.name);
-				buffer += "" + nature.name + " nature: ";
+				buffer += `${nature.name} nature: `;
 				if (nature.plus) {
-					const statNames = {'atk': "Attack", 'def': "Defense", 'spa': "Special Attack", 'spd': "Special Defense", 'spe': "Speed"};
-					buffer += "+10% " + statNames[nature.plus] + ", -10% " + statNames[nature.minus] + ".";
+					const statNames: {[k in StatNameExceptHP]: string} = {
```
Comment (Zarel) [inline server/chat-commands/info.ts:531]:
```suggestion
					const statNames = {
```

This annotation doesn't actually add anything...

Comment (KrisXV) [reply server/chat-commands/info.ts:531]: Have you tested this?

File: server/chat-commands/room-settings.ts (hunk 1 above)
Comment (Zarel) [reply server/chat-commands/room-settings.ts:1304]: I don't remember this at all. Can you move it to a separate PR now, though?

File: server/chat-commands/info.ts (hunk 3 above)
Comment (Zarel) [reply server/chat-commands/info.ts:216]: `curPunishment`

File: server/chat-commands/info.ts (hunk 5 above)
Comment (Zarel) [reply server/chat-commands/info.ts:531]: No. What error comes up?

Comment (KrisXV) [reply server/chat-commands/info.ts:531]: Nothing, I had it confused with another line

**Comment by KrisXV** [comment]
Done @Zarel

File: server/chat-commands/info.ts [hunk 6]
```diff
@@ -2396,7 +2496,7 @@ const commands = {
 		// XXX: target is trimmed by Chat#splitMessage. Let's not add another
 		// awful hack like ! or help command keys for whether or not the target
 		// is raw for now.
-		target = this.message.substr(this.cmdToken.length + this.cmd.length + +this.message.includes(' ')).trimEnd();
+		target = this.message.substr(this.cmdToken.length + this.cmd.length + +this.message.includes(' ')).trimRight();
```
Comment (Zarel) [inline server/chat-commands/info.ts:2499]: Why was this changed? According to MDN, the standard name is `trimEnd`.

File: server/chat-commands/info.ts [hunk 7]
```diff
@@ -2198,12 +2288,16 @@ const commands = {
 
 		Config.potd = target;
 		// TODO: support eval in new PM
-		Rooms.PM.eval('Config.potd = \'' + toID(target) + '\'');
+		(Rooms.PM as any).eval('Config.potd = \'' + toID(target) + '\'');
```
Comment (Zarel) [inline server/chat-commands/info.ts:2291]: Um, does this work at all? I would just comment it out if it doesn't...

File: server/chat-commands/info.ts (hunk 6 above)
Comment (KrisXV) [reply server/chat-commands/info.ts:2499]: according to VS code with an up-to-date package, trimEnd doesnt exist

File: server/chat-commands/info.ts (hunk 7 above)
Comment (KrisXV) [reply server/chat-commands/info.ts:2291]: i dont know, we havent used POTD in literal ages

Comment (KrisXV) [reply server/chat-commands/info.ts]: Just tested, can confirm it just causes crashes, so i commented it out

File: server/chat-commands/info.ts [hunk 8]
```diff
@@ -2198,12 +2288,16 @@ const commands = {
 
 		Config.potd = target;
```
Comment (Zarel) [inline server/chat-commands/info.ts:2289]:
```suggestion
		return this.errorReply(`Sorry, PotD is currently unsupported.`);

		Config.potd = target;
```

Comment (Zarel) [reply server/chat-commands/info.ts:2289]: I hope there's no lint rule against this. We might need to comment out _everything_ after the error.

Comment (KrisXV) [reply server/chat-commands/info.ts:2289]: A lint rule against.... putting inaccessible code in? I dont think this has never NOT been a lint rule

Comment (KrisXV) [reply server/chat-commands/info.ts]: I commented out the rest of the code and added the early return.

**Comment by Zarel** [comment]
Why did you end up deleting it entirely?

**Comment by Zarel** [comment]
I think it's useful to at least have an error message representing:

1. one day I will make it work again

2. if anyone tries to use it, they'll realize what happened to it

**Comment by Zarel** [comment]
Whatever, it's not really that big of a deal to me and I don't like leaving refactors uncommitted for so long.

**Comment by KrisXV** [comment]
I deleted it because it's still in the git history for when we decide to look into fixing it.

**Comment by Zarel** [comment]
Digging for it in the Git history (when there's not even a line to blame) is a _lot_ harder than looking at a comment.

**Comment by Zarel** [comment]
That's the advantage of the error message - it's a line that you can use Git Blame on to find the old commit.
